tests/test-location.c

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2019 Endless Mobile, Inc. */
+
+/* This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <glib.h>
+#include <glib-unix.h>
+
+#include "eins-location.h"
+
+static gboolean
+quit_main_loop (GMainLoop *main_loop)
+{
+  g_main_loop_quit (main_loop);
+  return G_SOURCE_REMOVE;
+}
+
+gint
+main (gint                argc,
+      const gchar * const argv[])
+{
+  GMainLoop *main_loop = g_main_loop_new (NULL, TRUE);
+  g_idle_add ((GSourceFunc) record_location_metric, "solutions-fake-image");
+  g_unix_signal_add (SIGTERM, (GSourceFunc) quit_main_loop, main_loop);
```

Comment:
I guess this is here because there is otherwise a race between this mock daemon closing itself and exiting cleanly, and the Python test calling `.terminate()` to kill it. But the test doesn't check the exit status of the mock daemon so is this needed?

Reply:
That's true, there's no reason why this needs to exit cleanly.